src/bot.py

```diff
@@ -53,7 +53,25 @@ def int_conv(string: str) -> bool:
     if args[1].startswith('u/'):
         args[1] = args[1][2:]
 
-    if int_conv(args[2]):
-        pass
-    elif args[2] in ('max', 'full', 'all'):
-        pass
+    # exception handling
+    # author too young and doesn't meet karma requirements
+    if int((time.time() - comment.author.created_utc) / (60 * 60 * 24)) < 9 and comment.author.link_karma + comment.author.comment_karma < 100:
+        comment.reply("ERROR_MESSAGE")
+        continue
+
+    # invalid gift arg
+    elif not int_conv(args[2]) or not args[2] in ("max", "full", "all"):
+        comment.reply("ERROR_MESSAGE")
+        continue
+       
+    # user gifting to != post author
+    elif comment.submission.author != args[1]:
+        comment.reply("ERROR_MESSAGE")
+        continue
+    
+    """
+    what needs to be done with the database:
+    elif int(args[2]) > author.vote_nugs:
+        comment.reply("ERROR MESSAGE")
```

Comment:
So, the only person users will be able to give nuggets to is the post author?

Reply:
That's how I thought we wanted it, otherwise you could just give nuggets to a random friend/alt and it wouldn't make sense.

Reply:
Good point. If that's the case, why not eliminate the second argument altogether?

Reply:
Ah true, I'd be all for that.